sdk/dotnet/Pulumi/Serialization/ResourcePackages.cs

```diff
@@ -106,7 +122,10 @@ private static IEnumerable<Assembly> LoadReferencedAssemblies()
                 }
             }
 
-            static bool PossibleMatch(AssemblyName? assembly) => assembly != null && !assembly.FullName.StartsWith("System", StringComparison.Ordinal);
+            static bool PossibleMatch(AssemblyName? assembly)
+                => assembly != null
+                && !assembly.FullName.StartsWith("System", StringComparison.Ordinal)
+                && assembly.ContentType != AssemblyContentType.WindowsRuntime;
```

Comment:
I believe I needed this condition because of the AspNetCore Framework reference. It was throwing trying to load `Windows.Storage` which is a WindowsRuntime assembly.